src/lib/shim/shim_api_addrinfo.c

```diff
@@ -368,7 +368,11 @@ int shim_api_getaddrinfo(const char* node, const char* service, const struct add
             }
         }
         // We've finished adding all relevant addresses.
-        return 0;
+        if (*res != NULL) {
+            return 0;
+        } else {
+            return EAI_NONAME;
```

Comment:
Could you add a test for this case to src/test/resolver/test_getaddrinfo.c?

Reply:
Added an IPv6 test